scrapli/transport/plugins/asynctelnet/transport.py

```diff
@@ -117,16 +117,25 @@ def _handle_control_chars(self) -> None:
         if not self.stdout:
             raise ScrapliConnectionNotOpened
 
+        if self._raw_buf.find(NULL) != -1:
```

Comment:
I added this to preserve the old behavior. If I understand the rest of the code, this isn't needed since `_read` catches `EOFError` and should be able to be removed? (`not c:` would only evaluate to true if c == 0?)